frontend/src/core/api/entity.js

```diff
@@ -141,18 +141,22 @@ export default class EntityAPI extends APIBase {
 
         const results = await this.fetch('/other-locales/', 'GET', payload, headers);
 
-        if (!Array.isArray(results)) {
+        if (typeof results !== 'object' || Array.isArray(results) || results == null) {
             return [];
         }
 
-        return results.map(entry => {
-            return {
-                code: entry.locale__code,
-                locale: entry.locale__name,
-                direction: entry.locale__direction,
-                script: entry.locale__script,
-                translation: entry.string,
-            };
-        });
+        const locales = Object.keys(results).map(key => results[key]).map(locale_lists => {
+            return locale_lists.map(entry => {
+                return {
+                    code: entry.locale__code,
+                    locale: entry.locale__name,
+                    direction: entry.locale__direction,
+                    script: entry.locale__script,
+                    translation: entry.string,
+                }
+            })
+        } );
```

Comment:
With https://github.com/mozilla/pontoon/pull/1508#discussion_r353735751, you can get rid of this entire block.

You'll need to update actions, reducer and component logic in the `otherlocales` module though, to take into account the new format of the `results` AKA `translations`.

Reply:
In changing the format type from Array to Object I keep coming up against errors regarding `inexact` values being incompatible with `exact OtherLocaleTranslation`. 

For example when I try changing the initial state of `translations` in reducer.js from an empty array to an empty object I get:  `Inexact object literal [1] is incompatible with exact OtherLocaleTranslation [2] in property translations.`

Should `results` not be set as an Object?

Reply:
Could you try defining `translations` like this?
```js
+translations: {
    preferred: Array<OtherLocaleTranslation>,
    other: Array<OtherLocaleTranslation>,
},
```

Now that I look at this, I'm thinking maybe we should rename preferred_translations -> preferred and other_translations -> other.

Reply:
That seems to have worked, but I did have to make that change in actions.js as well as for the Promise in entity.js as well.

The errors I'm getting now are all in relation to the format change with the exception of 
```javascript
if (results.status === false) {
            return null;
     }
```
which throws:  `Cannot return null because inexact null [1] is incompatible with exact
OtherLocaleTranslation [2] in type argument R`

Reply:
Glad to hear it worked. Well done!

You can either:
* return `{}` instead of null or
* use `null` as the default value of the `initialState`, in which case you should change `+translations: {` to `+translations: ?{` to make it [optional](https://flow.org/en/docs/types/maybe/).